ext/phar/phar.c

```diff
@@ -1226,14 +1226,15 @@ static zend_result phar_parse_pharfile(php_stream *fp, char *fname, size_t fname
 		/* if signature matched, no need to check CRC32 for each file */
 		entry.is_crc_checked = (manifest_flags & PHAR_HDR_SIGNATURE ? 1 : 0);
 		phar_set_inode(&entry);
-		// TODO: avoid copy
 		if (mydata->is_persistent) {
 			str = zend_string_init_interned(ZSTR_VAL(entry.filename), ZSTR_LEN(entry.filename), 1);
 		} else {
-			str = zend_string_init(ZSTR_VAL(entry.filename), ZSTR_LEN(entry.filename), 0);
+			str = entry.filename;
```

Comment:
The reason you don't increase the ref count is that you're transferring ownership, correct?

Reply:
No, the reason we don't increment the refcount is because the hash table code already does that for us.

Reply:
Ah right